packages/aws-cdk/lib/api/hotswap/ecs-services.ts

```diff
@@ -77,11 +77,17 @@ interface EcsService {
 
 class EcsServiceHotswapOperation implements HotswapOperation {
   public readonly service = 'ecs-service';
+  public readonly resourceNames: string[] = [];
 
   constructor(
     private readonly taskDefinitionResource: any,
     private readonly servicesReferencingTaskDef: EcsService[],
-  ) {}
+  ) {
+    for (const ecsService of servicesReferencingTaskDef) {
+      const serviceName: string = ecsService.serviceArn.split('/')[2];
+      this.resourceNames.push(serviceName);
+    }
```

Comment:
How about we use `map` here?

```suggestion
    this.resourceNames = servicesReferencingTaskDef.map(ecsService => ecsService.serviceArn.split('/')[2]);
```